Source/WebInspectorUI/UserInterface/Views/SearchSidebarPanel.js

```diff
@@ -95,8 +95,22 @@ WI.SearchSidebarPanel = class SearchSidebarPanel extends WI.NavigationSidebarPan
 
         this._inputElement.select();
 
-        if (performSearch)
-            this.performSearch(this._inputElement.value, {omitFocus: true});
+        if (performSearch) {
+            if (this._inputElement.value.length >= WI.SearchSidebarPanel.MinimumQueryLengthForSearchOnFocus)
+                this.performSearch(this._inputElement.value, {omitFocus: true});
+            else {
+                // Allow the user to intentionally proceed with a potentially very slow query by pressing the Enter key.
+                // This is needed because the "change" event won't trigger on Enter if the value is already set; there is no change.
+                let handleKeydownOnce = (event) => {
```

Comment:
since this is only used once we should just inline it as an argument to `addEventListener`